src/lib.rs

```diff
@@ -94,19 +130,26 @@ impl Poseidon {
 
     pub fn hash(&self, inp: Vec<Fr>) -> Result<Fr, String> {
         let t = inp.len() + 1;
-        if inp.is_empty() || inp.len() > self.constants.n_rounds_p.len() {
-            return Err("Wrong inputs length".to_string());
+        if inp.is_empty() {
+            return Err("Empty input".to_string());
         }
+        
+        // Check if we have the constants for this input length
+        let t_idx = t - 2;
+        if t_idx >= self.constants.c.len() {
+            return Err(format!("No constants available for input length {}. Enable the appropriate feature flag.", inp.len()));
+        }
```

Comment:
I tried using the lib externally, importing only the feature `poseidon-2`, and it breaks because the `self.constants.c.len()` is always `1` (same happens for the other poseidon sizes).

Reply:
Maybe a way to catch this for the future, might be splitting the test into a test for each size, and adding feature conditionals to the tests; something like
```rust
    #[test]
    #[cfg(feature = "poseidon-2")]
    fn test_poseidon2() {
        // ...
    }

    #[test]
    #[cfg(feature = "poseidon-3")]
    fn test_poseidon3() {
        // ...
    }
   // ...
```